src/ray/object_manager/plasma/plasma_allocator.h

```diff
@@ -19,53 +19,70 @@
 
 #include <cstddef>
 #include <cstdint>
+#include "ray/object_manager/plasma/allocator.h"
 
 namespace plasma {
 
-class PlasmaAllocator {
+// PlasmaAllocator that allocates memory from mmaped file to
+// enable memory sharing between processes.
+//
+// PlasmaAllocator is optimized for linux. On linux,
+// the Allocate call allocates memory from a pre-mmap file
+// from /dev/shm. On other system, it allocates memory from
+// a pre-mmap file on disk.
+//
+// The FallbackAllocate always allocates memory from a disk
+// based mmap file.
+class PlasmaAllocator : public IAllocator {
  public:
-  /// Allocates size bytes and returns a pointer to the allocated memory. The
-  /// memory address will be a multiple of alignment, which must be a power of two.
+  /// PlasmaAllocator can only be created once per process.
+  /// This is because it uses dlmalloc to allocate memory under the hood,
+  /// whose metadata is a global state(singleton).
+  static PlasmaAllocator &GetInstance();
+
+  /// On linux, it allocates memory from a pre-mmaped file from /dev/shm.
+  /// On other system, it allocates memory from a pre-mmaped file on disk.
+  /// return null if running out of space.
```

Comment:
```suggestion
  /// return null if running out of space.
```
This comment seems to be duplicated from the below line